e2e/specs/api/public.api.spec.ts

```diff
@@ -0,0 +1,476 @@
+import { TestSuite } from '../../../src/app/shared/models/testSuite';
+import { ProjectHelper } from '../../helpers/project.helper';
+import { TestRun } from '../../../src/app/shared/models/testRun';
+import { Test } from '../../../src/app/shared/models/test';
+import { TestResult } from '../../../src/app/shared/models/test-result';
+import { ATError } from '../../../src/app/shared/models/error';
+import { ApiAssertMessages, apiResponseErrors } from './api.constants';
+
+let suite: TestSuite;
+let test: Test;
+let testrun: TestRun;
+let testResult: TestResult;
+let atError: ATError;
+const unaccessibleId = 100000;
+const suiteName = 'My Suite';
+const editedSuiteName = 'My Edited Suite';
+const testName = 'My Test';
+const editedTestName = 'My Edited Test';
+const buildName = 'build_1';
+
+
+describe('Public API', () => {
+  const projectHelper: ProjectHelper = new ProjectHelper();
+
+  beforeAll(async () => {
+    await projectHelper.init();
+  });
+
+  afterAll(async () => {
+    await projectHelper.dispose();
+  });
+
+  afterEach(() => {
+    atError = undefined;
+  });
+
+  describe('Suite:', () => {
+    it('Can not create Suite via public API without name and id', async () => {
+      try {
+        await projectHelper.publicAPI.createOrUpdateSuite({
+          project_id: projectHelper.project.id
+        });
+      } catch (error) {
+        atError = error;
+      }
+
+      expect(atError).toBeDefined(ApiAssertMessages.errorNotRaised);
+      expect(atError.message).toBe(apiResponseErrors.missedIdOrName, ApiAssertMessages.errorIsWrong);
+    });
+
+    it('Can not create Suite via public API without project_id', async () => {
+      try {
+        await projectHelper.publicAPI.createOrUpdateSuite({
+          name: suiteName
+        });
+      } catch (error) {
+        atError = error;
+      }
+
+      expect(atError).toBeDefined(ApiAssertMessages.errorNotRaised);
+      expect(atError.message).toBe(apiResponseErrors.missedProjectId, ApiAssertMessages.errorIsWrong);
+    });
+
+    it('Can not create Suite via public API for unaccessible project', async () => {
+      try {
+        await projectHelper.publicAPI.createOrUpdateSuite({
```

Comment:
most of tests are the same, what if we create method like below and apply it where possible

  const checkApi = (callMethod: Function, expectedErrorMessage: String) => {
    try {
      callMethod();
    } catch (error) {
      atError = error;
    }
    expect(atError).toBeDefined(ApiAssertMessages.errorNotRaised);
    expect(atError.message).toBe(expectedErrorMessage, ApiAssertMessages.errorIsWrong);
  };

It could reduce nubber of code lines